File: training-data/ruby/gcd_lcm.rb
def gcd(a, b)
  a, b = b, a % b while b != 0
  a.abs
end

def lcm(a, b)
  return 0 if a.zero? || b.zero?
  (a * b).abs / gcd(a, b)
end

puts "#{gcd(48, 18)} #{lcm(4, 6)}"

File: training-data/rust/gcd_lcm.rs
fn gcd(mut a: i64, mut b: i64) -> i64 {
    while b != 0 {
        let t = b;
        b = a % b;
        a = t;
    }
    a.abs()
}

fn lcm(a: i64, b: i64) -> i64 {
    if a == 0 || b == 0 {
        0
    } else {
        (a * b).abs() / gcd(a, b)
    }
}

fn main() {
    println!("{} {}", gcd(48, 18), lcm(4, 6));
}
